modules/network/src/main/java/org/apache/ignite/network/DefaultMessagingService.java

```diff
@@ -115,6 +115,8 @@ public DefaultMessagingService(
         this.marshaller = marshaller;
 
         this.outboundExecutor = Executors.newSingleThreadExecutor(NamedThreadFactory.create(nodeName, "MessagingService-outbound-", LOG));
+        // TODO asch the implementation of delayed acks relies on absence of reordering on subsequent messages delivery.
```

Comment:
Because if it is not held the commit can be called with some write operation in progress.
But we need to enforce the following invariant on commit - all inflight operations are finished and all corresponding replication is completed.
Actually, looks like we can relax this invariant to "all corresponding replication is completed", because it implies that all updates on primary replica is also applied.